nemo/utils/exp_manager.py

```diff
@@ -564,12 +567,38 @@ class NeMoModelCheckpoint(ModelCheckpoint):
     """ Light wrapper around Lightning's ModelCheckpoint to force a saved checkpoint on train_end
     """
 
-    def __init__(self, save_best_model=False, postfix=".nemo", **kwargs):
+    def __init__(self, always_save_nemo=False, save_best_model=False, postfix=".nemo", **kwargs):
         # Parse and store "extended" parameters: save_best model and postfix.
+        self.always_save_nemo = always_save_nemo
         self.save_best_model = save_best_model
         self.postfix = postfix
         # Call the parent class constructor with the remaining kwargs.
         super().__init__(**kwargs)
+    
+    @rank_zero_only
+    def on_save_checkpoint(self, trainer, pl_module):
+        output = super().on_save_checkpoint(trainer, pl_module)
+         
+        if not self.always_save_nemo:
+            return output
+
+        # Load the best model and then re-save it
+        if self.save_best_model:
+            if not os.path.exists(self.best_model_path):
+                return output
+           
+            old_state_dict = deepcopy(pl_module.state_dict())
+            checkpoint = torch.load(self.best_model_path, map_location='cpu')
```

Comment:
Can we cache `best_model_path`, so that we can return if there was no change in best model?

Reply:
Yep that's a good idea.  I'll add a field in the NeMoModelCheckpoint to hold the previous one.